### PR TITLE
Added support for python3.12

### PR DESCRIPTION
**Pull request by gavin-aguiar**
Added support for python 3.12 for core tools

### Issue describing the changes in this PR

resolves #issue_for_this_pr

### Pull request checklist

* [ ] My changes **do not** require documentation changes
  * [ ] Otherwise: Documentation issue linked to PR
* [ ] My changes **do not** need to be backported to a previous version
  * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [ ] I have added all required tests (Unit tests, E2E tests)

### CODE REVIEW

**Review by liliankasem — DISMISSED** [review]

**Comment by surgupta-msft** [comment]
@gavin-aguiar Is this PR still relevant?

**Comment by gavin-aguiar** [comment]
> @gavin-aguiar Is this PR still relevant?

Yes, I'm waiting for the deployments to complete before merging this PR

File: src/Azure.Functions.Cli/Helpers/PythonHelpers.cs [hunk 1]
```diff
@@ -167,31 +167,31 @@ public static void AssertPythonVersion(WorkerLanguageVersionInfo pythonVersion,
...
             }
 
-            // Python 3.x (but not 3.[6|7|8|9|10|11]), not recommended, may fail. E.g.: 3.4, 3.5.
+            // Python 3.x (but not 3.[7|8|9|10|11|12]), not recommended, may fail. E.g.: 3.4, 3.5.
             if (pythonVersion.Major == 3)
             {
                 if (errorIfNotSupported)
-                    throw new CliException($"Python 3.6.x to 3.11.x is required for this operation. " +
-                        $"Please install Python 3.6, 3.7, 3.8, 3.9, 3.10 or 3.11 and use a virtual environment to switch to Python 3.6, 3.7, 3.8, 3.9, 3.10 or 3.11.");
-                ColoredConsole.WriteLine(WarningColor("Python 3.6.x, 3.7.x, 3.8.x, 3.9.x, 3.10.x or 3.11.x is recommended, and used in Azure Functions."));
```
Comment (vrdmr) [inline src/Azure.Functions.Cli/Helpers/PythonHelpers.cs:190]: We should add an aka.ms link to point to supported Python versions as well. And any documentation around it.

**Review by vrdmr — APPROVED** [review]

File: src/Azure.Functions.Cli/Helpers/PythonHelpers.cs (hunk 1 above)
Comment (gavin-aguiar) [reply src/Azure.Functions.Cli/Helpers/PythonHelpers.cs:190]: There are a few places where the doc needs to be updated. Will create a task and open a new PR

File: src/Azure.Functions.Cli/Helpers/PythonHelpers.cs [hunk 2]
```diff
@@ -167,31 +167,31 @@ public static void AssertPythonVersion(WorkerLanguageVersionInfo pythonVersion,
...
 
-            // Python 3.[6|7|8|9|10|11] (supported)
+            // Python 3.[7|8|9|10|11|12] (supported)
             if (IsVersionSupported(pythonVersion))
             {
                 return;
             }
 
-            // Python 3.x (but not 3.[6|7|8|9|10|11]), not recommended, may fail. E.g.: 3.4, 3.5.
+            // Python 3.x (but not 3.[7|8|9|10|11|12]), not recommended, may fail. E.g.: 3.4, 3.5.
```
[the remark this answers is not in the document]
Comment (gavin-aguiar) [reply src/Azure.Functions.Cli/Helpers/PythonHelpers.cs:184]: I don't think we need to keep updating the comment with every version update. This was just added as an example. Python 3.4 and 3.5 were never supported for azure functions, just added as an example

**Review by surgupta-msft — APPROVED** [review]

**Pull request merged**